### PR TITLE
🎨 clean up Factory pt 1

### PR DESCRIPTION
**Pull request by Thomas-Smets**

### CODE REVIEW

File: src/Factory.sol [hunk 1]
```diff
@@ -44,7 +44,7 @@ contract Factory is IFactory, ERC721, FactoryGuardian {
     struct AccountVersionInfo {
         address registry; // The contract address of the Registry.
         address logic; // The contract address of the Account logic.
-        bytes32 versionRoot; // The Merkle root of the merkle tree of all the compatible Account versions.
+        bytes32 versionRoot; // The Merkle root of the Merkle tree of all the compatible Account versions.
```
Comment (zekiblue) [inline src/Factory.sol:47]: Shouldn't this moved outside of the struct?

File: src/Factory.sol [hunk 2]
```diff
@@ -129,18 +129,20 @@ contract Factory is IFactory, ERC721, FactoryGuardian {
...
     function upgradeAccountVersion(address account, uint16 version, bytes32[] calldata proofs) external {
         if (_ownerOf[accountIndex[account]] != msg.sender) revert FactoryErrors.Only_Account_Owner();
-        if (accountVersionBlocked[version]) revert FactoryErrors.Account_Version_Blocked();
+        if (accountVersionBlocked[version]) revert FactoryErrors.AccountVersionBlocked();
         uint256 currentVersion = IAccount(account).ACCOUNT_VERSION();
 
         bool canUpgrade = MerkleProofLib.verify(
-            proofs, getAccountVersionRoot(), keccak256(abi.encodePacked(currentVersion, uint256(version)))
+            proofs,
+            accountDetails[latestAccountVersion].versionRoot,
```
Comment (zekiblue) [inline src/Factory.sol:144]: if its moved outside of struct needs to be used from general

File: src/Factory.sol [hunk 3]
```diff
@@ -206,7 +197,7 @@ contract Factory is IFactory, ERC721, FactoryGuardian {
      * @param from The sender.
      * @param to The target.
      * @param id The id of the Account that is about to be transferred.
-     * @dev This method overwrites the safeTransferFrom function in ERC721.sol to also transfer the Account proxy contract to the new owner.
+     * @dev This method overwrites the transferFrom function in ERC721.sol to also transfer the Account proxy contract to the new owner.
```
Comment (zekiblue) [inline src/Factory.sol:200]: do we need to mention as `ERC721.sol` can be `ERC721`?

File: src/Factory.sol (hunk 1 above)
Comment (Thomas-Smets) [reply src/Factory.sol:47]: Hold your horse, part 2 is coming!

File: src/AccountV1.sol [hunk 4]
```diff
@@ -169,7 +169,7 @@ contract AccountV1 is AccountStorageV1, IAccount {
             // If a creditor is set, new version should be compatible.
             // openMarginAccount() is a view function, cannot modify state.
             (bool success,,,) = ICreditor(creditor).openMarginAccount(newVersion);
-            if (!success) revert AccountErrors.Invalid_Account_Version();
+            if (!success) revert AccountErrors.InvalidAccountVersion();
```
Comment (j-vp) [inline src/AccountV1.sol:172]: top of file: no period after contract name

File: src/Factory.sol [hunk 5]
```diff
@@ -239,12 +231,10 @@ contract Factory is IFactory, ERC721, FactoryGuardian {
 
         if (IAccount(logic).ACCOUNT_VERSION() != latestAccountVersion) revert FactoryErrors.Version_Mismatch();
```
Comment (j-vp) [inline src/Factory.sol:232]: we can probably better cache latestAccountVersion here, but its an onlyOwner function

**Pull request merged**